.github/workflows/ci.yml

```diff
@@ -193,51 +186,31 @@ jobs:
 
       - id: push_cassettes
         name: Push updated cassettes
+        if: always() && (startsWith(github.event_name, 'pull_request') || success())
```

Comment:
we run this step in all cases if the previous steps are successful.

and if the previous steps are not successful, we still want the cassettes IF WE ARE IN A PR EVENT. Why ?
Because this way the tests will run way faster next time. It's very frustrating when a test takes 7 minutes to run and then it takes 7 minutes the next time too.